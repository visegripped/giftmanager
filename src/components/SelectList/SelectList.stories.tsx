
import React from 'react';
import { ComponentStory, ComponentMeta } from '@storybook/react';

import SelectList from './index';

// More on default export: https://storybook.js.org/docs/react/writing-stories/introduction#default-export
export default {
  title: 'component/SelectList',
  component: SelectList,
  // More on argTypes: https://storybook.js.org/docs/react/api/argtypes
} as ComponentMeta<typeof SelectList>;

// More on component templates: https://storybook.js.org/docs/react/writing-stories/introduction#using-args
const Template: ComponentStory<typeof SelectList> = (args) => <SelectList {...args} />;

export const BasicUsage = Template.bind({});
// More on args: https://storybook.js.org/docs/react/writing-stories/args
BasicUsage.args = {
  options : [],
  onSelect: () => {}
};
