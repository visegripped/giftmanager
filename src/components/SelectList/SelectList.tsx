import React from 'react';
import './SelectList.css';

interface OptionProps {
  value: string;
  label: string;
}

interface SelectListProps {
  options: OptionProps[];
  cssClasses: string;
  disabled: boolean;
  onChange(changeEvent: object, props: object): void;
}

const SelectList = (props: SelectListProps) => {
  const { options = [], cssClasses, disabled, onChange } = props;
  const selectChangeHandler = (changeEvent: React.ChangeEvent<HTMLSelectElement>) => {
    onChange(changeEvent, props);
  };
  const className = `selectList ${cssClasses}`;
  return (
    <select className={className} disabled={disabled} data-testid="SelectList" onChange={selectChangeHandler}>
      {options.map((option) => {
        const { value, label } = option;
        return (
          <option value={value} key={value}>
            {label || value}
          </option>
        );
      })}
    </select>
  );
};

export default SelectList;
