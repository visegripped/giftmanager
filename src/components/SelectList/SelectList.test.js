import { render, fireEvent, screen } from '@testing-library/react';
import SelectList from './index';

const sampleOptions = [
  { value: 'something', label: 'something' },
  { value: 'somethingElse', label: 'something else' },
  { value: 'somethingMore', label: 'something more' },
];

describe('SelectList', () => {
  test('onChange is executed properly', () => {
    let counter = 0;
    const testForClick = () => {
      counter += 1;
    };
    render(<SelectList options={sampleOptions} onChange={testForClick} />);
    const renderedSelectList = screen.getByTestId('SelectList');
    fireEvent.change(renderedSelectList, { target: { value: 1 } });
    expect(counter).toEqual(1);
  });
  test('snapshot matches', () => {
    render(<SelectList cssClasses="test classes" options={sampleOptions} disabled={false} selected="something" />);
    const renderedSelectList = screen.getByTestId('SelectList');
    expect(renderedSelectList).toMatchSnapshot();
  });
});
