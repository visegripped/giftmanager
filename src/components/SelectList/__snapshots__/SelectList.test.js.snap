// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SelectList snapshot matches 1`] = `
<select
  class="selectList test classes"
  data-testid="SelectList"
>
  <option
    value="someting"
  >
    something
  </option>
  <option
    value="sometingElse"
  >
    something else
  </option>
  <option
    value="sometingMore"
  >
    something more
  </option>
</select>
`;
